astropy/io/fits/scripts/fitsdiff.py

```diff
@@ -225,14 +225,14 @@ def match_files(paths):
 
     for path in paths:
         if glob.has_magic(path):
-            files = [os.path.abspath(f) for f in glob.glob(path)]
+            files = [os.path.abspath(os.path.join(path,f)) for f in glob.glob(path)]
```

Comment:
This cannot work if `path` is a glob pattern, and anyway `glob` returns the full path so there is no need to join the path here (a unit test will useful to check this).